modules/gapi/src/backends/fluid/gfluidcore.cpp

```diff
@@ -151,6 +151,242 @@ GAPI_FLUID_KERNEL(GFluidAddW, cv::gapi::core::GAddW, false)
+
+                vx_store(reinterpret_cast<uchar*>(&out[x]), v_pack_u(v_pack(v_round(a1 + b1), v_round(a2 + b2)),
+                                                                     v_pack(v_round(a3 + b3), v_round(a4 + b4))));
+            }
+
+            if (x < length && length >= nlanes)
+                x = length - nlanes;
+        }
+    }
+}
+
+template<typename T, typename VT>
+static inline void sub_simd_sametype(const T in1[], const T in2[], T out[], int length, int& x)
+{
+    constexpr int nlanes = VT::nlanes;
+    for (; x < length; )
+    {
+        for (; x <= length - nlanes; x += nlanes)
+        {
+            VT a = vx_load(&in1[x]);
+            VT b = vx_load(&in2[x]);
+            vx_store(&out[x], a - b);
+        }
+
+        if (x < length && length >= nlanes)
+            x = length - nlanes;
+    }
+}
+
+template<typename SRC, typename DST>
+static inline void sub_simd(const SRC in1[], const SRC in2[], DST out[], int length, int& x)
+{
+    if (std::is_same<DST, float>::value && !std::is_same<SRC, float>::value)
+        return;
+
+    if (std::is_same<DST, SRC>::value)
+    {
+        if (std::is_same<DST, uchar>::value)
+        {
+            sub_simd_sametype<uchar, v_uint8>(reinterpret_cast<const uchar*>(in1),
+                                              reinterpret_cast<const uchar*>(in2),
+                                              reinterpret_cast<uchar*>(out), length, x);
+        }
+        else if (std::is_same<DST, short>::value)
+        {
+            sub_simd_sametype<short, v_int16>(reinterpret_cast<const short*>(in1),
+                                              reinterpret_cast<const short*>(in2),
+                                              reinterpret_cast<short*>(out), length, x);
+        }
+        else if (std::is_same<DST, float>::value)
+        {
+            sub_simd_sametype<float, v_float32>(reinterpret_cast<const float*>(in1),
+                                                reinterpret_cast<const float*>(in2),
+                                                reinterpret_cast<float*>(out), length, x);
+        }
+    }
+    else if (std::is_same<SRC, short>::value && std::is_same<DST, uchar>::value)
+    {
+        constexpr int nlanes = v_uint8::nlanes;
+        for (; x < length; )
+        {
+            for (; x <= length - nlanes; x += nlanes)
+            {
+                v_int16 a1 = vx_load(reinterpret_cast<const short*>(&in1[x]));
+                v_int16 a2 = vx_load(reinterpret_cast<const short*>(&in1[x + nlanes / 2]));
+                v_int16 b1 = vx_load(reinterpret_cast<const short*>(&in2[x]));
+                v_int16 b2 = vx_load(reinterpret_cast<const short*>(&in2[x + nlanes / 2]));
+
+                vx_store(reinterpret_cast<uchar*>(&out[x]), v_pack_u(a1 - b1, a2 - b2));
+            }
+
+            if (x < length && length >= nlanes)
+                x = length - nlanes;
+        }
```

Comment:
move into specialization of `sub_simd_sametype`

Reply:
I don't see the sense. Please explain. Template sub_simd_sametype() was writed only for cases when both inputs type and output type are the same. In the case mentioned above input and output type are different and so required another handling.